### PR TITLE
Clause type

### LINKED ISSUE #126: add clause type to docs

### PR DESCRIPTION
**Pull request by jdville03**
Resolves #126 

Opening as a draft PR because we probably should wait for the bugs with the Clause type feature to get fixed before merging these updates.

_(There is a placeholder to include an image of a template draft view when the feature is fully working.)_

### CODE REVIEW

File: docs/markup-language/README.md [hunk 1]
```diff
@@ -968,6 +968,73 @@ The amount of federal income tax that must be withheld from the Employee's
 weekly wages is **$[[Amount of Income Tax Withheld]]**.
 ```
 
+## Clauses
+
+The Clause type in OpenLaw is equivalent to an embedded template. It can be used to insert pre-formed clauses, such as commonly used legal provisions (boilerplate)—whether simple or complex—into existing templates. The clause to be embedded in a template is technically a saved template itself and can include variables. When the clause is embedded in a template, those variables would render as part of the agreement text and have their own form fields as if they were directly included in that template's markup language.
```
Comment (outkaj) [inline docs/markup-language/README.md]: -> will render (a little stronger)

File: docs/markup-language/README.md [hunk 2]
```diff
@@ -968,6 +968,73 @@ The amount of federal income tax that must be withheld from the Employee's
...
+::: warning
+Note that since a Clause is a type of template, you must save that template separately before you try to embed it in another template, or else you will receive an error similar to below:
+
+```
+The template Choice of Law and Venue Clause could not be found on the server
+```
+
+:::
+
+The example markup language above will render the Draft view below. Note how the variables included in the embedded clause (the `Choice of Law and Venue Clause` template) are included as form fields in the `Simple Advisor Agreement` template.
```
Comment (outkaj) [inline docs/markup-language/README.md]: -> markup might be more succinct than "markup language" here

**Comment by jdville03** [comment]
Thanks @outkaj. I made those fixes. I'll let you know when this is updated with that image after the feature fixes are done.

**Comment by jdville03** [comment]
@outkaj this is ready for a final review. We can keep it as a draft PR until the feature is close to being ready to be deployed to the production instances.

File: docs/markup-language/README.md [hunk 3]
```diff
@@ -993,6 +962,65 @@ The amount of federal income tax that must be withheld from the Employee's
...
+
+<center>
+  <img src="./img/clause-type.png" alt="Embedded Clause" />
+</center>
+
+### Defining a Clause Type Variable
+
+When embedding a Clause type, you must give the variable a name as in the example above (`Choice of Law Insert`). Each clause embedded in a template should be defined with a name unique from any other variable defined in the template.
+
+The name of the variable can be useful to embed the clause more than once in the same template. For example, if you have already defined the clause in the template with `[[Choice of Law Insert: Clause("Choice of Law and Venue Clause")]]` you can render the same clause again in the template by just referencing `[[Choice of Law Insert]]`.
```
Comment (outkaj) [inline docs/markup-language/README.md]: comma needed after )]]

Comment (jdville03) [reply docs/markup-language/README.md]: thanks - fixed now.

**Pull request merged**